api/app.py

```diff
@@ -69,14 +78,26 @@ async def ValidateFile(file: UploadFile = File(), override_levels: list[int] = F
 
 @app.post("/execute/duckdb/", status_code=status.HTTP_200_OK)
 async def ExecuteBackend(data: list[str]):
-    global con
-    return ExecuteDuckDb(data, con)
+    global duckConn
+    return ExecuteDuckDb(data, duckConn)
 
 
 @app.post("/parse/", status_code=status.HTTP_200_OK)
 async def ParseToSubstrait(data: dict):
-    global con
-    return ParseFromDuckDB(data, con)
+    global duckConn
+    return ParseFromDuckDB(data, duckConn)
+
+
+@app.post("/save/", status_code=status.HTTP_200_OK)
+async def SavePlan(data: PlanData):
+    response = await mongoConn.add_record(data)
+    return response
+
+
+@app.post("/fetch/", status_code=status.HTTP_200_OK)
+async def FetchPlan(id: str):
+    response = await mongoConn.get_record(id)
+    return response["json_data"], response["validation_levels"]
```

Comment:
Instead of an array with two elements would it make more sense to return a JSON object?

```
{
  "json_data": ...,
  "validation_levels": ...
}
```

Reply:
made the change, thanks for the suggestion.